app/models/oral_history_content/date_and_place.rb

```diff
@@ -8,5 +8,22 @@ class DateAndPlace
     attr_json :province, :string
     attr_json :country, :string
 
+    def update_from_hash(hsh)
+      self.date =         hsh['date']
+      self.city =         hsh['city']
+      self.state =        hsh['state']
+      self.province =     hsh['province']
+      self.country =      hsh['country']
+    end
```

Comment:
I don't think you should need this, I am pretty sure .new should do the same thing, you should really be able to do `model.new(hash)` instead of `model.update_from_hash(hash)`, let's debug why not if not.

Reply:
You were absolutely right -- just using a simple data.values().all? { |x| x.empty? } in the controller was fine.

Reply:
I think that comment about empty? is about something other than `update_from_hash`?

I have absolutely nothing against providing a `def empty?` that does that, instead of having to do `values.all` every time, a `def empty?` in the model is a nice solution! Either way is fine. 

If you do want to do that line, style is not use parens after the method call, and there's a shortcut for `{|x| x.empty?}`, so it can look like this: `data.values.all?(:empty?)`.

Reply:
Fixed!